### PR TITLE
Examples for improving maven_install.json and maven_install

### PR DESCRIPTION
**Pull request by TerrorJacktyl**
I opened this PR to provide supporting examples [for this issue.](https://github.com/bazelbuild/rules_jvm_external/issues/897) Please check that out for more sense and context :))

### CODE REVIEW

File: examples/spring_boot/WORKSPACE [hunk 1]
```diff
@@ -32,7 +32,8 @@ maven_install(
     fetch_sources = True,
     maven_install_json = "//:maven_install.json",
     repositories = [
-        "https://repo1.maven.org/maven2",
+        "https://repo1.error.maven.org/maven2",
```
Comment (TerrorJacktyl) [inline examples/spring_boot/WORKSPACE:35]: I've simulated a flake by just changing the first URL to an erroneous one.

File: examples/spring_boot/maven_install.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "__AUTOGENERATED_FILE_DO_NOT_MODIFY_THIS_FILE_MANUALLY": "THERE_IS_NO_DATA_ONLY_ZUUL",
-  "__INPUT_ARTIFACTS_HASH": 278723230,
-  "__RESOLVED_ARTIFACTS_HASH": 1635567545,
+  "__INPUT_ARTIFACTS_HASH": 433939764,
+  "__RESOLVED_ARTIFACTS_HASH": 385007839,
```
Comment (TerrorJacktyl) [inline examples/spring_boot/maven_install.json]: Note that the lockfile changed even though the file contents didn't (ie the actual checksums of the third-party code).

File: examples/spring_boot/maven_install.json [hunk 3]
```diff
@@ -1,1483 +1,1459 @@
 {
-    "dependency_tree": {
```
Comment (shs96c) [inline examples/spring_boot/maven_install.json:2]: You appear to have repinned this file using an older version of `rules_jvm_external`. Could you please rebase on `HEAD` of `master` and repin again?

**Comment by TerrorJacktyl** [comment]
Hey @shs96c ! I didn't get a ping on this sorry so it went unnoticed.

This PR isn't actually a fix for anything, it merely contains examples to illustrate some problems with the current lockfile generation. As of when I opened this PR, these problems were still evident in the v2 lockfile generation too. [Please see this issue for context](https://github.com/bazelbuild/rules_jvm_external/issues/897).

**Comment by TerrorJacktyl** [comment]
This PR should not be reviewed, it only serves to support https://github.com/bazelbuild/rules_jvm_external/issues/897.

**Pull request closed without merging**